lib/specs.js

```diff
@@ -36,8 +29,11 @@ function assembleProtocol(fpath, opts, cb) {
   if (!opts.importHook) {
     opts.importHook = files.createImportHook();
   }
+  if (!opts.relativeResolveHook) {
+    opts.relativeResolveHook = files.createRelativeResolveHook();
+  }
```

Comment:
I tried moving this functionality into the import hook, passing in both the import name *and* the name of the file it's being imported from, but it turned out to be more complex than expected. In particular, the "import idl from nested paths" test has some tricky behavior--say we have the following IDLs:

```
'a/1.avdl': 'import idl "2.avdl"; protocol A1 { fixed One(1); }',
'a/2.avdl': 'import idl "../b/3.avdl"; protocol A2 { fixed Two(2); }',
'b/3.avdl': 'protocol B3 { fixed Three(3); }'
```

When we import from `a/1.avdl`, we pull in `a/2.avdl`. Unfortunately, it only knows its relative name, as imported from `a/1.avdl`, which will be `"2.avdl"`. When we then import `b/3.avdl` from `a/2.avdl`, we need to resolve `b/3.avdl` *relative* to `a/2.avdl`, but `a/2.avdl` doesn't "know" that it's in the `a` directory--recall that its name is just `"2.avdl"` because that's what it was imported as.

There needs to be some way to pass along the "resolved so far" path of `a/2.avdl`, which was resolved when it was imported from `a/1.avdl`. This could be done through the import hook, by passing both the imported file's contents *and* its resolved path into the callback. Does that behavior sound good?

Reply:
I've reverted the file path changes for now. I'll revisit them in another PR.